app/src/features/apiClient/screens/apiClient/utils.ts

```diff
@@ -184,3 +198,92 @@ export const convertFlatRecordsToNestedRecords = (records: RQAPI.Record[]) => {
   return updatedRecords;
 };
 export const getEmptyPair = (): KeyValuePair => ({ id: Math.random(), key: "", value: "", isEnabled: true });
+
+export const updateAuthOptions = (
+  entry: RQAPI.Entry,
+  currentAuthType: AUTHORIZATION_TYPES,
+  updatedKey?: string,
+  updatedValue?: string,
+  formValues = {}
+) => {
+  const currentEntry = { ...entry };
+
+  const currentFormValues = {
+    ...formValues[currentAuthType],
+    ...(updatedKey || updatedValue ? { [updatedKey]: updatedValue } : {}),
+  };
+
+  currentEntry.auth = {
+    currentAuthType,
+    [currentAuthType]: currentFormValues,
+  };
+
+  return currentEntry;
+};
+
+export const deleteAuthOptions = (dataToUpdate: KeyValuePair[]) => {
+  return dataToUpdate.filter((option) => option.type !== AUTH_ENTRY_IDENTIFIER);
+};
+
+export const updateRequestWithAuthOptions = (
+  dataToUpdate: KeyValuePair[],
+  dataToAppend: RQAPI.AuthOptions,
+  dataAppendType: AUTHORIZATION_TYPES
+) => {
+  const dataToUpdateCopy = !isEmpty(dataToUpdate)
+    ? [...deleteAuthOptions(JSON.parse(JSON.stringify(dataToUpdate)))]
+    : JSON.parse(JSON.stringify(dataToUpdate));
+
+  let newKeyValuePair: KeyValuePair;
+
+  const createAuthorizationHeader = (type: string, key: string, value: string): typeof newKeyValuePair => ({
+    id: Math.random(),
+    key,
+    value,
+    type,
+    isEnabled: true,
+  });
+
+  const updateDataInState = (data: KeyValuePair[], type = AUTH_ENTRY_IDENTIFIER, key: string, value: string) => {
+    const existingIndex = data.findIndex((header) => header.type === type);
+    const newHeader = createAuthorizationHeader(type, key, value);
+
+    if (existingIndex !== -1) {
+      data[existingIndex] = { ...data[existingIndex], ...newHeader };
+    } else {
+      data.unshift(newHeader);
+    }
+  };
+
+  switch (dataAppendType) {
+    case AUTHORIZATION_TYPES.NO_AUTH:
+      break;
+    case AUTHORIZATION_TYPES.BASIC_AUTH: {
+      const { username, password } = dataToAppend[dataAppendType];
+      updateDataInState(
+        dataToUpdateCopy,
+        AUTH_ENTRY_IDENTIFIER,
+        "Authorization",
+        `Basic ${btoa(`${username || ""}:${password || ""}`)}`
+      );
+      break;
+    }
+    case AUTHORIZATION_TYPES.BEARER_TOKEN: {
+      const { bearer } = dataToAppend[dataAppendType];
+      updateDataInState(dataToUpdateCopy, AUTH_ENTRY_IDENTIFIER, "Authorization", `Bearer ${bearer}`);
+      break;
+    }
+    case AUTHORIZATION_TYPES.API_KEY: {
+      const { key, value } = dataToAppend[dataAppendType];
+
+      if (key) {
+        updateDataInState(dataToUpdateCopy, AUTH_ENTRY_IDENTIFIER, key || "", value || "");
+      }
+      break;
+    }
+    default:
+      break;
+  }
+
+  return dataToUpdateCopy;
+};
```

Comment:
- Abstract this into a separate file `helpers/auth.ts`.

Reply:
Also we should update this method definition so that it can support showing the auth headers in the headers tab too. Something like this
```
function processAuth(auth: AuthOptions): {
  headers: { key: string; value: string }[]; // KeyValuePair // These are the auth headers that will be added to the request,
  params: { key: string; value: string }[]; // KeyValuePair, // These are the query parameters that will be added to the request
} {}

```

- Then we can consume these return values in `executeAPIRequest` and call a util method `updateRequestAuthHeaders` and `updateRequestQueryParams`
- And we can call the same method processAuth in the Headers Tab and use those values to render the new headers added by the auth